resdk/resources/collection.py

```diff
@@ -28,11 +28,14 @@ class BaseCollection(BaseResource):
     WRITABLE_FIELDS = ('data', 'description', 'settings', 'descriptor_schema',
```

Comment:
It may be confusing that getting the data would return hydrated objects whereas setting them expects ids. I suggest not to allow saving (to add or remove) data through the `data` attribute. Maybe we should even make this a method, i.e. `data()`. We should then support adding and removing data through helper functions, e.g. `add_data` and `remove_data`.